docs/connector-development/cdk-python/schemas.md

```diff
@@ -25,3 +25,64 @@ def get_json_schema(self):
     return schema
 ```
 
+## Schema normalization
+
+It is important to ensure output data conforms to the declared json schema. This is because the destination receiving this data to load into tables may strictly enforce schema (e.g. when data is stored in a SQL database, you can't put INTEGER type into CHAR column). In the case of changes to API output (which is almost guaranteed to happen over time) or a minor mistake in jsonschema definition, data syncs could thus break because of mismatched datatype schemas.
+
+To remain robust in operation, the CDK provides a transformation ability to perform automatic object mutation to align with desired schema before outputting to the destination. All streams inherited from airbyte_cdk.sources.streams.core.Stream class have this transform configuration available. It is _disabled_ by default and can be configured per stream within a source connector.
+### Default schema normalization
+Lets say you want have output records from controller to be cast to the type described on json schema. This is how you can configure it:
+
+```python
+from airbyte_cdk.sources.utils.transform import TransformConfig, Transformer
+from airbyte_cdk.sources.streams.core import Stream
+
+class MyStream(Stream):
+    ...
+    transformer = Transformer(TransformConfig.DefaultSchemaNormalization)
+    ...
+```
+In this case default transformation will be applied. For example if you have schema like this
+```json
+{"type": "object", "properties": {"value": {"type": "string"}}}
+```
+and source API returned object with non-string type, it would be casted to string automaticaly:
+```json
+{"value": 12} -> {"value": "12"}
+```
+Also it works on complex types:
+```json
+{"value": {"unexpected_object": "value"}} -> {"value": "{'unexpected_object': 'value'}"}
```

Comment:
in this example the types are unchanged right?

Reply:
object {"unexpected_object": "value"} has been casted to a string representing object **"**{'unexpected_object': 'value'}**"**

Reply:
AH I see it now. thanks!